### DISCUSSION TITLE
Cursorline colors

### DISCUSSION THREAD

**Discussion by Ha5hBr0wn** [helix-editor/helix#10427] [Ideas]
Hi,

I am new to using helix and so far I have really liked it, but something that I am missing from Vim is the ability to have a different color cursor line for when i am in Insert mod vs Normal mode. 

For example I want my cursor line to have a blue background when I am in Normal mode, and a red background when I am in insert mode.

I find it a lot easier than looking to the bottom to see what mode I am in.

Adding such a feature would be really nice. 

Thanks

**Comment by gabydd** [comment]
I think originally this wasn't done as to not overcomplicate the theme scopes https://github.com/helix-editor/helix/pull/2170#issuecomment-1114256994, to implement this though you would probably want to do something similar to the matching here https://github.com/helix-editor/helix/pull/5130/files though with just the ``ui.cursorline.secondary`` and ``ui.cursorline.primary`` prefix

**Comment by danillos** [comment]
Currently, we can do it on the cursor only:

```
"ui.cursor"                   = { fg = "dark-choco-60", bg = "white-choco-10" }

"ui.cursor.insert"            = { fg = "dark-choco-60", bg = "red10" }
"ui.cursor.primary.insert"    = { fg = "dark-choco-60", bg = "red10" }

"ui.cursor.select"            = { fg = "dark-choco-60", bg = "blue10" }
"ui.cursor.primary.select"    = { fg = "dark-choco-60", bg = "blue10" }
```

**Comment by Ha5hBr0wn** [comment]
Do I just add this to the .toml file of the theme I am using?

**Comment by gabydd** [comment]
to add that to the theme you are using, you can make a new theme file in ``~/.config/helix/themes``  and then use [inheritance](https://docs.helix-editor.com/themes.html#inheritance) like so:
```toml
inherits = "<name of theme you are using>"
# theme scopes you want to override
```
then use the name of that new file as your theme name

this is separate to this issue though and still doesn't solve having a different cursorline color depending on the mode

**Comment by Ha5hBr0wn** [comment]
```
inherits = "solarized_light"

"ui.cursor"                   = { fg = "dark-choco-60", bg = "white-choco-10" }

"ui.cursor.insert"            = { fg = "dark-choco-60", bg = "red10" }
"ui.cursor.primary.insert"    = { fg = "dark-choco-60", bg = "red10" }

"ui.cursor.select"            = { fg = "dark-choco-60", bg = "blue10" }
"ui.cursor.primary.select"    = { fg = "dark-choco-60", bg = "blue10" }

```
I put this in a file and set my theme to it. But all that happens is when I enter insert, or select mode my cursor disappears. Also I don't want to affect select mode. I want blue on normal mode if that is possible. I understand that it won't give me a full cursor line but even just changing the cursor color would still be pretty good. Thanks

**Comment by gabydd** [comment]
something like this will make the primary cursor blue in normal mode and keep it the same as the theme in normal and select (the above is the palette from a different theme which is probably why your cursor is disappearing)
```toml
inherits = "solarized_light"

"ui.cursor.primary.normal" = { fg = "base03", bg = "blue" }
```

**Comment by Ha5hBr0wn** [comment]
Ok that fg setting works for everything. Thanks. I will leave the issue open because I still think it would be a nice feature to implement
